src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRazorViewEngineFileProviderAccessor.cs

```diff
@@ -18,7 +19,15 @@ public class DefaultRazorViewEngineFileProviderAccessor : IRazorViewEngineFilePr
         public DefaultRazorViewEngineFileProviderAccessor(IOptions<RazorViewEngineOptions> optionsAccessor)
         {
             var fileProviders = optionsAccessor.Value.FileProviders;
-            if (fileProviders.Count == 1)
+            if (fileProviders.Count == 0)
+            {
+                var message = Resources.FormatFileProvidersAreRequired(
+                    typeof(RazorViewEngineOptions).FullName,
+                    nameof(RazorViewEngineOptions.FileProviders),
+                    typeof(IFileProvider).FullName);
+                throw new ArgumentException(message, nameof(optionsAccessor));
```

Comment:
1. If this list of file providers is empty, executing any `ViewResult` throws an `InvalidOperationException` stating it couldn't find the specified view in one or more searched locations. I.e. the behaviour is identical to the case where the file is actually missing and the error doesn't point users toward the cause.
2. This provider is used only in code directly related to finding and compiling views. And the `Exception` bubbles up correctly. Agree that we previously had issues with less-than-useful (i.e. totally obfuscating) errors due to exceptions in service constructors but that's no longer the case.
3. Can't check for an empty provider collection in the consuming code due to back-compat: Must leave `IRazorViewEngineFileProviderAccessor` unchanged and down-casting to check for a new `DefaultRazorViewEngineFileProviderAccessor` property (e.g. `public int Count { get; }`) is just ugly.